expression/constant.go

```diff
@@ -28,17 +28,21 @@ import (
 
 // NewOne stands for a number 1.
 func NewOne() *Constant {
+	retT := types.NewFieldType(mysql.TypeTiny)
```

Comment:
I do not understand why we need to change these 2 funcs

Reply:
because i make test fail when i modify AggFieldType. The test is palnner/cascades/transformation_rules_test.go  TestTransformAggToProj.  "SQL": "select count(b), sum(b), avg(b), b, max(b), min(b), bit_and(b), bit_or(b), bit_xor(b) from t group by a having sum(b) >= 0 and count(b) >= 0 order by b",

The reason for fail is that the bit_or will add a cast wrap  (cast(b as bigint unsigned binary), the arg is unsigned.
the TransformAggToProj  will add ifnull wrap ifnull(cast(b as bigint unsigned binary), expression.NewZero()).
ifnull use AggFieldType infer type to tigger integral promotion.
[rule_aggregation_elimination.go](https://github.com/pingcap/tidb/blob/master/planner/core/rule_aggregation_elimination.go)  rewriteBitFunc

I think this is a easy way to process. what do you think?